lib/puppet/module/task.rb

```diff
@@ -157,7 +193,14 @@ def implementations
     end
 
     def files
-      implementations.map {|imp| { 'name' => imp['name'], 'path' => imp['path'] } }
+      md = metadata
+      unless md.nil?
+        lib_files = md.dig('files').nil? ? [] : self.class.find_files(md['files'], @module)
+        impl_lib_files = md.dig('implementations', 'files').nil? ? [] : self.class.find_files(md['implementations']['files'], @module)
```

Comment:
Files is going to be an entry in each implementation. It's also possible it's duplicated in several implementations. So digging for `implementations.files` doesn't seem like it will cover that.

Reply:
i.e. `implementations` is an array, not an object, so this will never find anything.

Reply:
Yep, apologies, fixed, though I don't call `uniq` on the final files array. I'll  add that.